script.transmission/resources/lib/gui.py

```diff
@@ -232,32 +274,47 @@ def close(self):
 
 
 class TorrentInfoGUI(xbmcgui.WindowXMLDialog):
-    def __init__(self, strXMLname, strFallbackPath, strDefaultName, bforeFallback=0):
+    def __init__(self,
+                 strXMLname,
+                 strFallbackPath,
+                 strDefaultName,
+                 bforeFallback=0):
         self.transmission = None
         self.torrent_id = None
         self.list = {}
         self.timer = None
+
     def setTorrent(self, transmission, t_id):
         self.transmission = transmission
         self.torrent_id = t_id
         self.timer = threading.Timer(UPDATE_INTERVAL, self.updateTorrent)
         self.timer.start()
+
     def updateTorrent(self):
         pbar = self.getControl(219)
         list = self.getControl(220)
         labelName = self.getControl(1)
         labelStatus = self.getControl(2)
         labelProgress = self.getControl(11)
-        torrent = self.transmission.info()[self.torrent_id]
+
+        # Catch KeyError which indicates the torrent has been removed by user
+        try:
+            torrent = self.transmission.info()[self.torrent_id]
+        except KeyError:
+            return
+
         files = self.transmission.get_files(self.torrent_id)[self.torrent_id]
 
         statusline = "[%(status)s] %(down)s down, %(up)s up (Ratio: %(ratio).2f)" % \
```

Comment:
If statusline is used displayed to the user in the GUI, it should also be localized.